src/lib.rs

```diff
@@ -2325,6 +2345,33 @@ impl Build {
         .map(|x| x.to_owned()))
     }
 
+    /// Some platforms have multiple, compatible, canonical prefixes. Look through
+    /// each possible prefix for a compiler that exists and return it. The prefixes
+    /// should be ordered from most-likely to least-likely.
+    fn find_working_gnu_prefix(&self, prefixes: &[&'static str]) -> Option<&'static str> {
+        let suffix = if self.cpp { "-g++" } else { "-gcc" };
+
+        for prefix in prefixes {
+            let mut cmd = Command::new(&format!("{}{}", prefix, suffix));
```

Comment:
Instead of spawning a process could this perhaps probe `PATH` which should be a bit speedier?

Reply:
I considered that, but then I'd have to check all `PATHEXT` on Windows for each possible executable extension, and check for the `+x` bit on Posix systems.

One concern I had here was what happens if the compiler expects something on `stdin` and never actually finishes, but I imagine that's a largely theoretical problem.

I'm also not sure how well case sensitivity works here. But I'm sure Rust's `Path::exists()` does the right thing there.

If you prefer, I can rework it.

Reply:
My main worry is that process spawning is pretty slow compared to checking for files. There's not need to really be 100% correct here since it's pretty unlikely you'd have multiple of these compilers in `PATH` but only one is actually executable. And AFAIK but most of these compilers don't even exist on Windows, right?

Reply:
At least two of these compilers exist on Windows (`riscv-embed-none` and `riscv64-unknown-none-elf`) in compiled form, so I could imagine users running into both of these.

I've rewritten it so it now iterates through the `PATH` variable rather than forking off commands. Furthermore, it seems like `Path::exists()` is case-insensitive on Windows, so it continues to work even though I tried renaming my compiler to `riscv64-unknown-elf-gcc.EXE`.